mousestyles/powerlaw/kolmogorov_test.py

```diff
@@ -0,0 +1,160 @@
+
+    Parameters
+    ----------
+    strain: int {0, 1, 2}
+        The strain of mouse to test
+    mouse: int {0, 1, 2, 3}
+        The mouse twin id with in the strain
+    day: int {0, 1, ..., 11}
+        The day to calculate the distance
+    distribution: A Scipy Stats Continuous Distribution
+        {stats.pareto, stats.expon, stats.gamma}
+        The distribution to test against. Currently support pareto, expon, and
+        gamma, but any one-sided continuous distribution in Scipy.stats should
+        work.
+    is_plot: boolean
+        If True, will plot the PDF and CDF for visualization
+    verbose: boolean
+        If True, will print out testing result
+
+    Returns
+    -------
+    params: np.ndarray shape (p,)
+        The optimal parameter for the distribution. Optimal in the sense of
+        minimizing K-S statistics.
+    The function also print out the Kolmogorov-Smirnov test result for three
+    cases
+    1. When comparing the empirical distribution against the distribution with
+    parameters estimated with MLE
+    2. When comparing the empirical distribution against the distribution with
+    parameters estimated by explicitely minimizing KS statistics
+    3. When comparing a resample with replacement of the empirical distribution
+    against the Pareto in 2.
+    A p-value > 0.05 means we fail to reject the Null hypothesis that the
+    empirical distribution follow the specified distribution.
+
+    Notes:
+    ------
+    The MLE often does not fit the data very well. We instead minimizing the
+    K-S distance, and obtain a better fit (as seen by the PDF and CDF
+    similarity between sample data and the fit)
+
+    References:
+    -----------
+        1. Kolmogorov-Smirnov test:
+            https://en.wikipedia.org/wiki/Kolmogorov-Smirnov_test
+        2. Pareto Distribution (also known as power law distribution)
+            https://en.wikipedia.org/wiki/Pareto_distribution
+
+    Examples:
+    ---------
+    >>> res = perform_kstest(0, 0, 0, is_plot=False, verbose=False)
+    >>> np.allclose(res, np.array([3.67593246, 0.62795748, 0.37224205]))
+    True
+    """
+    dist_name = distribution.__class__.__name__.split("_")[0]
+    # Get distances travelled in 20ms for this strain, this mouse, on this day
+    df = load_movement(strain=strain, mouse=mouse, day=day)
+    x = np.array(np.sqrt(df.x.diff()**2 + df.y.diff()**2))[1:]
+    x = x[x >= 1]
+
+    # Fit the parameters by MLE
+    mle_params = distribution.fit(x)
+
+    # Fit the Parameter by explicitely minimize KS-Statistics, and perform
+    # K-S test.
+    def calculate_ks_statistics(params):
+        """This will return the kolmogorov test statistics of the empirical
+        distribution of x and the pareto distribution with parameter params.
+        """
+        return stats.kstest(x, dist_name, args=params)[0]
+
+    if dist_name == "pareto":
+        # A good initial estimate for Pareto parameter
+        init_params = [4.5, 0.5, 0.5]
```

Comment:
Sure. so distribution is a scipy.stats.rv_continuous object, and the distribution.fit will fit maximum likelihood parameter to the data. See for example scipy doc [http://docs.scipy.org/doc/scipy-0.14.0/reference/generated/scipy.stats.rv_continuous.fit.html](url)